drivers/dma/dma_stm32u5.c

```diff
@@ -188,6 +188,14 @@ void stm32_dma_enable_stream(DMA_TypeDef *dma, uint32_t id)
 	LL_DMA_EnableChannel(dma, dma_stm32_id_to_stream(id));
 }
 
+bool stm32_dma_is_enabled_stream(DMA_TypeDef *dma, uint32_t id)
+{
+	if (LL_DMA_IsEnabledChannel(dma, dma_stm32_id_to_stream(id)) == 1) {
+		return true;
+	}
+	return false;
+}
+
 int stm32_dma_disable_stream(DMA_TypeDef *dma, uint32_t id)
```

Comment:
In this function you have a call to LL_DMA_IsEnabledChannel (line 207), that could be changed to use your new function.
Sorry for not putting this comment under the specific line directly, but apparently Github does not allow leaving a comment to a line too far from a modification...

Reply:
true, but...
it keeps the driver like it is done in the dma_stm32_v1 or dma_stm32_v2  and that will probably save few cpu instructions at runtime.

Reply:
@erwango should the driver stm32_dma_disable_stream call the stm32_dma_is_enabled_stream
here and also in _dma_stm32_v1.c_ or _dma_stm32_v2.c_ ?

Reply:
I think we should try to make it simple.
Dedicated U5 driver was done in order to simplify things. I don't think we should add complexity to get all drivers look similar.